src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/config/ServiceConfigItem.java

```diff
@@ -62,5 +75,26 @@ public void setAllowCreatingCaseBeforePaymentsAreProcessed(boolean allowCreating
         this.allowCreatingCaseBeforePaymentsAreProcessed = allowCreatingCaseBeforePaymentsAreProcessed;
     }
 
+    public String getSurnameOcrFieldName(String formType) {
+        return formTypeToSurnameOcrFieldMappings.get(formType);
+    }
+
+    public void setFormTypeToSurnameOcrFieldMappings(List<FormFieldMapping> formTypeToSurnameOcrFieldMappings) {
+        this.formTypeToSurnameOcrFieldMappings = formTypeToSurnameOcrFieldMappings.stream()
+            .collect(groupingBy(FormFieldMapping::getFormType, toList()))
+            .entrySet().stream()
+            .collect(toMap(
+                e -> e.getKey(),
+                e -> {
+                    if (e.getValue().size() > 1) {
+                        log.error("Form type {} has {} mappings to surname fields",
+                            e.getKey(),
+                            e.getValue().size());
+                    }
+                    return e.getValue().get(0).getOcrField();
+                }
+            ));
```

Comment:
how about instead of double streaming and collecting we do in single go?

```java
this.formTypeToSurnameOcrFieldMappings = formTypeToSurnameOcrFieldMappings.
    .stream()
    .collect(
        HashMap::new,
        (map, mapping) -> {
            String formType = mapping.getFormType();
            if (map.containsKey(formType)) {
                log.error(
                    "Duplicate form type mapping found for {} form and value {}. Using first one: {}", formType,
                    formType,
                    mapping.getOcrField(),
                    map.get(formType)
                );
            } else {
                map.put(formType, mapping.getOcrField());
            }
        },
        HashMap::putAll
    );
```

Reply:
I expected some comment like this. Ok

Reply:
Given intolerance to duplicates, I guess we can simplify this quite a bit.

Reply:
Yes, then generic `Collectors.toMap` would work - it checks for duplicates by definition/default